controllers/application/ingress_virtual_service.go

```diff
@@ -71,8 +73,120 @@ func (r *ApplicationReconciler) reconcileIngressVirtualService(ctx context.Conte
 	return reconcile.Result{}, err
 }
 
-// Filter for virtual services named like *-ingress
-func isIngressVirtualService(virtualService *networkingv1beta1.VirtualService) bool {
-	match, _ := regexp.MatchString("^.*-ingress$", virtualService.Name)
-	return match
+func (r *ApplicationReconciler) defineRedirectVirtualService(ctx context.Context, application *skiperatorv1alpha1.Application) (*networkingv1beta1.VirtualService, error) {
+	virtualService := networkingv1beta1.VirtualService{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      application.Name + "-http-redirect",
+			Namespace: application.Namespace,
+		},
+		Spec: networkingv1beta1api.VirtualService{
+			ExportTo: EXPORT_TO_NAMESPACES,
+			Gateways: r.getGatewaysFromApplication(application),
+			Hosts:    []string{"*"},
+			Http: []*networkingv1beta1api.HTTPRoute{
+				{
+					Match: []*networkingv1beta1api.HTTPMatchRequest{
+						{
+							WithoutHeaders: map[string]*networkingv1beta1api.StringMatch{
+								":path": {
+									MatchType: &networkingv1beta1api.StringMatch_Prefix{
+										Prefix: "/.well-known/acme-challenge/",
+									},
+								},
+							},
+							Port: 80,
+						},
+					},
+					Redirect: &networkingv1beta1api.HTTPRedirect{
+						Scheme:       "https",
+						RedirectCode: 302,
```

Comment:
```suggestion
						RedirectCode: 308,
```

> Although the [specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301#specifications) requires the method and the body to remain unchanged when the redirection is performed, not all user-agents meet this requirement. Use the 301 code only as a response for [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET) or [HEAD](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD) methods and use the [308 Permanent Redirect](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308) for [POST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST) methods instead, as the method change is explicitly prohibited with this status.

See MDN on [301](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301) and [308](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/308).